packages/myst-frontmatter/src/contributors/contributors.yml

```diff
@@ -166,3 +166,133 @@ cases:
             family: Author
             given: Test
     errors: 1
+  - title: editor/reviewer added to collaborators
+    raw:
+      reviewer: Jane Doe
+      editor: John Doe
+    normalized:
+      reviewers:
+        - Jane Doe
+      editors:
+        - John Doe
+      contributors:
+        - id: Jane Doe
+          name: Jane Doe
+          nameParsed:
+            literal: Jane Doe
+            family: Doe
+            given: Jane
+        - id: John Doe
+          name: John Doe
+          nameParsed:
+            literal: John Doe
+            family: Doe
+            given: John
+  - title: editor/reviewer collaborators pass
+    raw:
+      reviewers:
+        - janedoe
+      editors:
+        - johndoe
+      contributors:
+        - id: janedoe
+          name: Jane Doe
+          nameParsed:
+            literal: Jane Doe
+            family: Doe
+            given: Jane
+        - id: johndoe
+          name: John Doe
+          nameParsed:
+            literal: John Doe
+            family: Doe
+            given: John
+    normalized:
+      reviewers:
+        - janedoe
+      editors:
+        - johndoe
+      contributors:
+        - id: janedoe
+          name: Jane Doe
+          nameParsed:
+            literal: Jane Doe
+            family: Doe
+            given: Jane
+        - id: johndoe
+          name: John Doe
+          nameParsed:
+            literal: John Doe
+            family: Doe
+            given: John
+  - title: editor/reviewer authors pass
+    raw:
+      reviewers:
+        - janedoe
+      editors:
+        - johndoe
+      authors:
+        - id: janedoe
+          name: Jane Doe
+          nameParsed:
+            literal: Jane Doe
+            family: Doe
+            given: Jane
+        - id: johndoe
+          name: John Doe
+          nameParsed:
+            literal: John Doe
+            family: Doe
+            given: John
+    normalized:
+      reviewers:
+        - janedoe
+      editors:
+        - johndoe
+      authors:
+        - id: janedoe
+          name: Jane Doe
+          nameParsed:
+            literal: Jane Doe
+            family: Doe
+            given: Jane
+        - id: johndoe
+          name: John Doe
+          nameParsed:
+            literal: John Doe
+            family: Doe
+            given: John
+  - title: author as id reference resolves to object
```

Comment:
This edge case triggered some changes to the `stash` functionality. Previously, a warning was raised on author `janedoe`, referenced by id here. Even though `janedoe` is defined in `reviewers` and all the final frontmatter resolution was correct, initial validation still raised a warning since "janedoe" does not have a given name when parsed...

In this case, it's just an ID, not a name, so we don't actually want that warning, but we _would_ want the warning if it never resolved as an ID - then we'd have to assume it is in fact a name with no given name.

This means the `validateAndStash` functionality needs to suppress some warnings, but then maybe re-raise them later... Pretty tedious, and probably doesn't really matter. But now this tests pass as well as all the old stash-related tests.